metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/parser/AnalysisFileTypeFromSet.java

```diff
@@ -0,0 +1,157 @@
+    private static final Logger logger = LoggerFactory.getLogger(AnalysisFileTypeFromSet.class);
+    private static final String propertiesFile = "application.properties";
+
+    @Override
+    public List<AnalysisFileType> extract(AnalysisSetType analysisSet) {
+        List<AnalysisFileType> subAnalysisFileList = new ArrayList<>();
+
+        AnalysisType[] analysisType = analysisSet.getANALYSISArray();
+        for (AnalysisType analysis : analysisType) {
+            AnalysisType.FILES analysisFILES = analysis.getFILES();
+            AnalysisFileType[] analysisFile = analysisFILES.getFILEArray();
+            for (AnalysisFileType anaFile : analysisFile) {
+                subAnalysisFileList.add(anaFile);
+            }
+        }
+        return subAnalysisFileList;
+    }
+
+    public AnalysisSetType getAnalysisSet(String xmlStr) throws XmlException {
+        return ANALYSISSETDocument.Factory.parse(xmlStr).getANALYSISSET();
+    }
+
+    public List<AnalysisFileType> extractFromSqlXml(SQLXML sqlXml) {
+        String xmlStr;
+        List<AnalysisFileType> analysisFileList = new ArrayList<>();
+
+        try {
+            xmlStr = sqlXml.getString();
+            AnalysisSetType analysisSet;
+            try {
+                analysisSet = getAnalysisSet(xmlStr);
+                analysisFileList = extract(analysisSet);
+            } catch (XmlException e) {
+                logger.info("Unable to convert XML String to AnalysisSet: ", xmlStr);
+            }
+        } catch (SQLException e) {
+            logger.info("Unable to convert SQLXML Object to String: ", sqlXml.toString());
+        }
+        return analysisFileList;
+    }
+
+    private Properties loadPropertiesFile() throws IOException {
+        Properties properties = new Properties();
+        InputStream inFile = null;
+        try {
+            inFile = new FileInputStream(propertiesFile);
+            try {
+                properties.load(inFile);
+            } catch (IOException e) {
+                logger.info("Unable to load JDBC connection properties file");
+                throw e;
+            }
+        } catch (FileNotFoundException e) {
+            logger.info("JDBC connection properties file not found: " + propertiesFile);
+            throw e;
+        } finally {
+            if (inFile != null) {
+                inFile.close();
+            }
+        }
+        return properties;
+    }
+
+    private String createUrl(Properties appProperties) {
+        String url = appProperties.getProperty("OracleJDBC.url");
+        String host = appProperties.getProperty("OracleJDBC.host");
+        String protocol = appProperties.getProperty("OracleJDBC.protocol");
+        String serviceName = appProperties.getProperty("OracleJDBC.servicename");
+        String port = appProperties.getProperty("OracleJDBC.port");
+        url = url + ":@(DESCRIPTION =(ADDRESS_LIST =(ADDRESS = (PROTOCOL = " + protocol + ")(HOST = " + host + ")(PORT = " +
```

Comment:
Url can be made in the properties file i guess

Reply:
Well, i gave options on all parameters in properties file. Let me know, i can make full URL there

Reply:
done